lib/move/index.js

```diff
@@ -35,7 +35,9 @@ function move (source, dest, options, callback) {
   }
 
   function doRename () {
-    if (overwrite) {
+    if (path.resolve(source) === path.resolve(dest)) {
+      setTimeout(callback, 1)
+    } else if (overwrite) {
```

Comment:
@jdalton what are your thoughts on `setTimeout` over `setImmediate`? It's my understanding `setTimeout` with an input of 1, still clamps to 15 ms.